cpp/src/round/round.cu

```diff
@@ -47,102 +52,190 @@ T __device__ generic_abs(T value)
 }
 
 template <typename T, typename std::enable_if_t<std::is_signed<T>::value>* = nullptr>
-bool __device__ is_negative(T value)
+int16_t __device__ generate_sign(T value)
 {
-  return value < 0;
+  return value > 0 ? 1 : value < 0 ? -1 : 0;
```

Comment:
This is weird.  Sign cannot be zero.

```suggestion
  return value < 0 ? -1 : 1;
```

Previously you used `auto const sign = is_negative(e) ? -1 : 1;` -- no zero.

Reply:
That is true, but I thought what you were advocating for what is known as the [`signum` function](http://www.cs.cas.cz/portal/AlgoMath/MathematicalAnalysis/SpecialFunctions/SignumFunction.htm), similar to what APL has:
```
      × 42 0 ¯1729 ⍝ monadic × in APL is "direction" (aka the sign of the number)
1 0 ¯1
```

Reply:
No I'm advocating something that replaces `auto const sign = is_negative(e) ? -1 : 1;`